Agents/DashboardAgent/agent/src/main/java/uk/ac/cam/cares/jps/agent/dashboard/DashboardAgent.java

```diff
@@ -0,0 +1,189 @@
+        // If it can be converted to second, return run time in seconds
+        if (TimeUnit.NANOSECONDS.toSeconds(duration) > 0) {
+            jsonMessage.accumulate("Runtime", TimeUnit.NANOSECONDS.toSeconds(duration) + "s");
+        } else {
+            // Else return as nanoseconds
+            jsonMessage.accumulate("Runtime", duration + "ns");
+        }
+        return jsonMessage;
+    }
+
+    /**
+     * Validates the request parameters.
+     *
+     * @return true or false depending on valid parameter status.
+     */
+    @Override
+    public boolean validateInput(JSONObject requestParams) {
+        return true;
+    }
+
+    /**
+     * Run logic for the "/status" route that indicates the agent's current status.
+     *
+     * @return A response to the request called as a JSON Object.
+     */
+    protected JSONObject statusRoute() {
+        JSONObject response = new JSONObject();
+        LOGGER.info("Detected request to get agent status...");
+        if (DashboardAgent.VALID) {
+            response.put("Result", "Agent is ready to receive requests.");
+        } else {
+            response.put("Error", "Agent could not be initialised! Please ensure it is running on a stack!");
+        }
+        return response;
+    }
+
+    /**
+     * Run logic for the "/setup" route to set up the dashboards on the stack.
+     *
+     * @return A response to the request called as a JSON Object.
+     */
+    protected JSONObject setupRoute() {
+        JSONObject response = new JSONObject();
+        if (DashboardAgent.VALID) {
+            LOGGER.info("Setting up client to interact with dashboard...");
+            try {
+                DashboardClient client = new DashboardClient(SERVICES);
+                client.initDashboard();
+            } catch (Exception e) {
+                LOGGER.fatal("Dashboard could not be set up! " + e.getMessage());
+                response.put("Error", "Dashboard could not be set up! " + e.getMessage());
+                return response;
+            }
+            LOGGER.info("Dashboard has been successfully set up!");
+            response.put("Result", "Dashboard has been successfully set up!");
+        } else {
+            response.put("Error", "Agent could not be initialised! Please check logs for more information...");
+        }
+        return response;
+    }
+
+    /**
+     * Run logic for the "/reset" route to reset the agent and re-execute all queries from the knowledge graph.
+     *
+     * @return A response to the request called as a JSON Object.
+     */
+    protected JSONObject resetRoute() {
+        JSONObject response = new JSONObject();
+        if (DashboardAgent.VALID) {
+            SERVICES = new StackClient();
+            response.put("Result", "Agent has been successfully reset!");
+        } else {
+            response.put("Error", "Agent could not be initialised! Please check logs for more information...");
+        }
+        return response;
+    }
```

Comment:
What is this useful for? As far as I can tell if `VALID` is `true` then `SERVICES` should also be initialised.